pkg/controllermanager/controller/shoot/state/finalizer/finalizer_suite_test.go

```diff
@@ -0,0 +1,17 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package finalizer
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestShootState(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ControllerManager Controller ShootState Finalizer Test Suite")
```

Comment:
```suggestion
	RunSpecs(t, "ControllerManager Controller Shoot State Finalizer Test Suite")
```

(according to directory structure)

However, why is `finalizer` a subpackage? Please move this into `controller/shoot/shootstate` (or even `controller/shootstate` since it's mainly reconciling `ShootState`s and not `Shoot`s) to make it more clear.

Reply:
We initially tried to follow the `controllerregistration` folder structure for the seed finalizer controller: `controllermanager/controller/controllerregistration/seedfinalizer`. Tbh I don't remember why we left it with `shoot/state/finalizer` instead of `shootstatefinalizer` 😅

Reply:
Thanks for taking the time to review the structure, let me know if you find the latest version suitable:

* `/pkg/controllermanager/controller/shootstate/` - containing the controller's logic as suggested in https://github.com/gardener/gardener/pull/11491#discussion_r1982839091
* `/test/integration/controllermanager/shootstate` - containing the controller's integration tests

I was considering having 

* `/pkg/controllermanager/controller/shootstate/finalizer/` - where to keep the controller sources
* `/pkg/controllermanager/controller/shootstate/add.go` - where to register the controller with the manager, much like what the rest of the controllers do

Please let me know your thoughts on it :))